### PR TITLE
[libc][newheadergen]: adding entry_point testing

### PR DESCRIPTION
**Pull request by aaryanshukla**

### CODE REVIEW

File: libc/newhdrgen/tests/expected_output/test_header.h [hunk 1]
```diff
@@ -40,10 +40,6 @@ _Float16 func_c(int, float) __NOEXCEPT;
 _Float16 func_d(int, float) __NOEXCEPT;
 #endif // LIBC_TYPES_HAS_FLOAT16
 
-#ifdef LIBC_TYPES_HAS_FLOAT16_AND_FLOAT128
```
Comment (michaelrj-google) [inline libc/newhdrgen/tests/expected_output/test_header.h:43]: would it make more sense to leave this function here and add a new function for the entrypoint testing?

Comment (aaryanshukla) [reply libc/newhdrgen/tests/expected_output/test_header.h]: fixed, let me if I'm good to merge

**Comment by jhuber6** [comment]
I'm somewhat wondering what the actual performance implications are for the trivial implementation versus the `realloc` one. I.e.
```
size_t chars = vsnprintf(nullptr, 0, format, args)
void *buffer = malloc(chars + 1);
vsnprintf(buffer, chars, format, args);
```
Just wondering since the GPU really hates function pointers.

**Review by michaelrj-google — APPROVED** [review]
LGTM, good to land

**Pull request merged**